projecttacoma/fqm-execution
# Data requirements comparison script

# Summary
Since we are reworking our data requirements calculation in fqm-execution, we have found a need for comparing our data requirements output to that of the [elm-parser-for-ecqms](https://github.com/projecttacoma/elm-parser-for-ecqms), which we know to be a more accurate data requirements calculation.

## New behavior
No need behavior within fqm-execution, just the addition of scripts for comparison and the `/data-requirements` directory that contains everything (structured similarly to the `/regression` directory).

## Code changes
- `data-requirements/README.md` - explains the purpose of the data-requirements directory and scripts and details instructions on how to run the scripts and how to interpret their outputs.
- `data-requirements/compare.sh` - this script compares the overall outputs of the data requirements (similar to the regression script)
- `.gitignore` - add some directories from `/data-requirements`
- `.gitkeep` - add `jan-2024-connectathon` to keep an empty directory
- `data-requirements/fqm-e-dr.ts` - helper file for calculating the data-requirements for all measures in jan-2024-connectathon using fqm-execution (used in both bash scripts)
- `data-requirements/summary-compare.sh` - summary comparison bash script
- `data-requirements/summary-compare.ts` - summary comparison helper file

# Testing guidance
NOTE: I detail most of the necessary information in the `/data-requirements/README.md` so please let me know if there is anything missing in that file that would be helpful!
- First, read through the `README.md` for running instructions and output descriptions.
- Make sure to get the january-2024-connectathon files from me that we cannot check into this repository. 
- Make sure both of the scripts run and that their outputs make sense.
- I am mainly looking for feedback on the readability and usability of these scripts. I spent potentially too much time on this task trying to figure out what outputs were the most helpful. In an effort to keep things moving, I want to get feedback on the current state. I am of the opinion that if there are any major suggestions (i.e. "We should output the comparison results to their own files!") that we should make a backlog task for them.
